src/components/walletConnectModal.tsx

```diff
@@ -3,25 +3,14 @@ import React from 'react';
 import MyAlgoConnectButton from './myAlgoConnectButton';
 import WalletConnectButton from './walletConnectButton';
 import * as buffer from 'buffer';
-import useWalletAccount from '../utils/persistAccount';
 
 if (typeof window !== `undefined`) {
   (window as any).Buffer = buffer.Buffer;
 }
 
-const truncate = (address: string) => address.substring(0, 4) + "..." + address.substring(address.length-4, address.length)
-
 const WalletConnectModal = () => {
   const { isOpen, onOpen, onClose } = useDisclosure()
-  const [account, setAccount] = useWalletAccount();
 
-  const handleDisconnect = () => {
-    setAccount({});
-    localStorage.setItem('walletconnect', '{}');
-  }   
-  if (account?.address) {
-    return <Button onClick={handleDisconnect}>{truncate(account?.address)}</Button>
-  }
   return (
     <>
       <Button onClick={onOpen}>Connect Wallet</Button>
```

Comment:
pulled disconnect and the "button when account is connected" component out of walletConnectModel and into their own component.